### ISSUE TITLE
SCDF dashboard UI - Can't open Streams or Tasks with the latest 2.x Version

### ISSUE THREAD

**Issue by ilayaperumalg** [spring-cloud/spring-cloud-dataflow-ui#1000]
_From @barisa on December 4, 2018 9:40_

**Description:**
When trying to add a Stream or Task via the dashboard UI I can't open the `Streams` neither `Tasks` module and get the following error _No property DEFINITION found for type StreamDefinition!_ (see Screenshot).

Error log:

```
dataflow-server    | 2018-12-04 09:32:13.167 ERROR 1 --- [nio-9393-exec-1] o.s.c.d.s.c.RestControllerAdvice         : Caught exception while handling a request
dataflow-server    |
dataflow-server    | org.springframework.data.mapping.PropertyReferenceException: No property DEFINITION found for type StreamDefinition!
dataflow-server    | 	at org.springframework.data.mapping.PropertyPath.<init>(PropertyPath.java:94)
dataflow-server    | 	at org.springframework.data.mapping.PropertyPath.create(PropertyPath.java:358)
dataflow-server    | 	at org.springframework.data.mapping.PropertyPath.create(PropertyPath.java:334)
dataflow-server    | 	at org.springframework.data.mapping.PropertyPath.lambda$from$0(PropertyPath.java:287)
dataflow-server    | 	at java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:324)
dataflow-server    | 	at org.springframework.data.mapping.PropertyPath.from(PropertyPath.java:269)
dataflow-server    | 	at org.springframework.data.mapping.PropertyPath.from(PropertyPath.java:252)
dataflow-server    | 	at org.springframework.data.jpa.repository.query.QueryUtils.toJpaOrder(QueryUtils.java:563)
dataflow-server    | 	at org.springframework.data.jpa.repository.query.QueryUtils.toOrders(QueryUtils.java:516)
dataflow-server    | 	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.getQuery(SimpleJpaRepository.java:630)
dataflow-server    | 	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.getQuery(SimpleJpaRepository.java:584)
dataflow-server    | 	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll(SimpleJpaRepository.java:387)
dataflow-server    | 	at org.springframework.data.jpa.repository.support.SimpleJpaRepository.findAll(SimpleJpaRepository.java:357)
dataflow-server    | 	at sun.reflect.GeneratedMethodAccessor129.invoke(Unknown Source)
dataflow-server    | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
dataflow-server    | 	at java.lang.reflect.Method.invoke(Method.java:498)
dataflow-server    | 	at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:377)
dataflow-server    | 	at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:200)
dataflow-server    | 	at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:629)
```

**Release versions:**
```
{
  "versionInfo": {
    "implementation": {
      "name": "spring-cloud-dataflow-server-local",
      "version": "2.0.0.BUILD-SNAPSHOT"
    },
    "core": {
      "name": "Spring Cloud Data Flow Core",
      "version": "2.0.0.BUILD-SNAPSHOT"
    },
    "dashboard": {
      "name": "Spring Cloud Dataflow UI",
      "version": "1.7.0.RELEASE"
    },
    "shell": {
      "name": "Spring Cloud Data Flow Shell",
      "version": "2.0.0.BUILD-SNAPSHOT",
      "url": "https://repo.spring.io/libs-snapshot/org/springframework/cloud/spring-cloud-dataflow-shell/2.0.0.BUILD-SNAPSHOT/spring-cloud-dataflow-shell-2.0.0.BUILD-SNAPSHOT.jar"
    }
  },
  "featureInfo": {
    "analyticsEnabled": true,
    "streamsEnabled": true,
    "tasksEnabled": true,
    "schedulerEnabled": false
  },
  "securityInfo": {
    "isAuthenticationEnabled": false,
    "isAuthorizationEnabled": false,
    "isFormLogin": false,
    "isAuthenticated": false,
    "username": null,
    "roles": []
  },
  "runtimeEnvironment": {
    "appDeployer": {
      "platformSpecificInfo": {},
      "deployerImplementationVersion": "1.1.2.RELEASE",
      "deployerName": "Spring Cloud Skipper Server",
      "deployerSpiVersion": "2.0.0.BUILD-SNAPSHOT",
      "javaVersion": "1.8.0_192",
      "platformApiVersion": "",
      "platformClientVersion": "",
      "platformHostVersion": "",
      "platformType": "Skipper Managed",
      "springBootVersion": "2.0.4.RELEASE",
      "springVersion": "5.0.8.RELEASE"
    },
    "taskLauncher": {
      "platformSpecificInfo": {},
      "deployerImplementationVersion": "1.3.9.RELEASE",
      "deployerName": "LocalTaskLauncher",
      "deployerSpiVersion": "1.3.4.RELEASE",
      "javaVersion": "1.8.0_192",
      "platformApiVersion": "Linux 4.9.125-linuxkit",
      "platformClientVersion": "4.9.125-linuxkit",
      "platformHostVersion": "4.9.125-linuxkit",
      "platformType": "Local",
      "springBootVersion": "2.0.4.RELEASE",
      "springVersion": "5.0.8.RELEASE"
    }
  }
}
```

**Steps to reproduce:**
```
$ wget https://raw.githubusercontent.com/spring-cloud/spring-cloud-dataflow/master/spring-cloud-dataflow-server-local/docker-compose.yml

$ export DATAFLOW_VERSION=latest
$ export SKIPPER_VERSION=1.1.2.RELEASE
$ docker-compose up
```

**Screenshots:**
<img width="1142" alt="screenshot 2018-12-04 at 10 29 10" src="https://user-images.githubusercontent.com/1219881/49432887-f0bca600-f7b0-11e8-9bca-c24e3ed9d9c8.png">

**Additional context:**
Creating and deploying streams with the SCDF shell works.

_Copied from original issue: spring-cloud/spring-cloud-dataflow#2646_

**Comment by ilayaperumalg** [comment]
@barisa The recent changes from SCDF core requires changes in the sorting parameter names at the client side. Hence, the SCDF UI is updated with the PR #1001. Once this PR is merged, you can see the SCDF UI fine for stream/task definitions pages. Thanks again for reporting this!

**Comment by barisa** [comment]
@ilayaperumalg thanks for the quick fix.

The stream and task panel are working now.
But I see a quite similar problem for the **_Jobs_** panel.
Any Ideas? 

## Error Log:

```
018-12-05 20:21:18.410 ERROR 1 --- [io-9393-exec-10] o.s.c.d.s.c.RestControllerAdvice         : Caught exception while handling a request
d-flow           |
d-flow           | org.springframework.jdbc.BadSqlGrammarException: StatementCallback; bad SQL grammar [SELECT E.JOB_EXECUTION_ID, E.START_TIME, E.END_TIME, E.STATUS, E.EXIT_CODE, E.EXIT_MESSAGE, E.CREATE_TIME, E.LAST_UPDATED, E.VERSION, I.JOB_INSTANCE_ID, I.JOB_NAME FROM BATCH_JOB_EXECUTION E, BATCH_JOB_INSTANCE I WHERE E.JOB_INSTANCE_ID=I.JOB_INSTANCE_ID ORDER BY JOB_EXECUTION_ID DESC LIMIT 30]; nested exception is org.postgresql.util.PSQLException: ERROR: relation "batch_job_execution" does not exist
d-flow           |   Position: 170
d-flow           | 	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:234)
```

## Screenshot
<img width="1066" alt="screenshot 2018-12-05 at 21 18 54" src="https://user-images.githubusercontent.com/1219881/49541587-7b95c180-f8d3-11e8-8471-6619a5c17e78.png">

**Comment by ilayaperumalg** [comment]
@barisa I couldn't replicate this issue on the latest snapshot. I tried this on `local` server with embedded H2  database. Could you share more info on your environment (database etc.,)?

**Comment by sabbyanandan** [comment]
@barisa: Please make sure you're testing on the [latest snapshot build](http://docs.spring.io/spring-cloud-dataflow/docs/2.0.0.BUILD-SNAPSHOT/reference/htmlsingle/#getting-started-deploying-spring-cloud-dataflow). I'll close this issue; if you continue to see the anomaly, feel free to open another issue.
